platform/viewer/cypress/support/commands.js

```diff
@@ -30,12 +30,73 @@ Cypress.Commands.add('openStudy', (patientName) => {
     cy.visit('/');
     cy.get('#patientName')
       .type(patientName);
-    cy.get('.studylistStudy > .patientName')
-      .contains(patientName)
-      .click();
+
+    const patientResult = '.studylistStudy > .patientName';
+
+    cy.get(patientResult)
+      .then({ timeout: 5000 }, ($patientResult) => {
+        cy.contains(patientName)
+        .click();
+      })
   }
 );
 
 Cypress.Commands.add('drag', {prevSubject: 'element',},
   (...args) => DragSimulator.simulate(...args)
 );
+
+// Command to perform two clicks into two different positions. Each position must be (x, y).
+// The positions are considering the element as reference, therefore, top-left of the element will be (0, 0).
+Cypress.Commands.add('addLine', (element, initPosition, finalPosition) =>  {
+
+  cy.get(element)
+    .click(initPosition[0], initPosition[1], { force: true })
+    .then(() =>{
+      cy.get(element)
+        .trigger('mousemove', { clientX: finalPosition[0], clientY: finalPosition[1] })
+        .click({ force: true })
+    })
+});
```

Comment:
To illustrate the points made below with `addAngle`, here is how you might rewrite this method:

```js
/**
 * Command to perform two clicks into two different positions. Each position must be [x, y].
 * The positions are considering the element as reference, therefore, top-left of the element will be (0, 0).
 *
 * @param {*} viewport - Selector for viewport we would like to interact with
 * @param {number[]} firstClick - Click position [x, y]
 * @param {number[]} secondClick - Click position [x, y]
 */
Cypress.Commands.add('addLine', (viewport, firstClick, secondClick) =>  {

  cy.get(viewport).then(($viewport) => {
    const [ x1, y1 ] = firstClick;
    const [ x2, y2 ] = secondClick;

    $viewport.click(x1, y1, { force: true});
    $viewport.click(x2, y2, { force: true });
  });
});
```

The goal here is readability. Functionally, these should accomplish the same thing. We're focused on trying to make the code read more like common language, or simple steps; as well as add clarity around the types/shape of the parameters required for this command.

Reply:
Thanks for all the great suggestions!